### PR TITLE
Handle close_notify in ConnectionPoolHandler

### PR DESCRIPTION
**Pull request by jguerra**
I also added a new status category to differentiate this failure from connection resets

### CODE REVIEW

File: zuul-core/src/main/java/com/netflix/zuul/exception/OutboundErrorType.java [hunk 1]
```diff
@@ -55,6 +55,10 @@ public enum OutboundErrorType implements ErrorType {
             ERROR_TYPE_ORIGIN_RESET_CONN_STATUS.get(),
             ZuulStatusCategory.FAILURE_ORIGIN_RESET_CONNECTION,
             ClientException.ErrorType.CONNECT_EXCEPTION),
+    CLOSE_NOTIFY_CONNECTION(
+            504,
```
Comment (argha-c) [inline zuul-core/src/main/java/com/netflix/zuul/exception/OutboundErrorType.java:59]: nit: This is technically the equivalent of `CONNECT_ERROR` imo, which maps to a `502` vs `504`

File: zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/ConnectionPoolHandler.java [hunk 2]
```diff
@@ -97,6 +99,11 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
                         + reason.name() + ", " + ChannelUtils.channelInfoForLogging(ctx.channel());
                 closeConnection(ctx, msg);
             }
+        } else if(evt instanceof SslCloseCompletionEvent event) {
```
Comment (argha-c) [inline zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/ConnectionPoolHandler.java:102]: 👍🏼

File: zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/ConnectionPoolHandler.java [hunk 3]
```diff
@@ -58,6 +60,7 @@ public ConnectionPoolHandler(OriginName originName) {
         this.inactiveCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_inactive", originName.getMetricId());
         this.errorCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_error", originName.getMetricId());
         this.headerCloseCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_headerClose", originName.getMetricId());
+        this.sslCloseCompletionCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_sslClose", originName.getMetricId());
```
Comment (argha-c) [inline zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/ConnectionPoolHandler.java:63]: 🙏🏼

**Review by argha-c — APPROVED** [review]

File: zuul-core/src/main/java/com/netflix/zuul/exception/OutboundErrorType.java (hunk 1 above)
Comment (jguerra) [reply zuul-core/src/main/java/com/netflix/zuul/exception/OutboundErrorType.java]: Updated 👍

**Pull request merged**